go/arrow/array/list.go

```diff
@@ -146,28 +152,196 @@ func (a *List) Release() {
 	a.values.Release()
 }
 
-type ListBuilder struct {
+func (a *List) ValueOffsets(i int) (start, end int64) {
+	debug.Assert(i >= 0 && i < a.array.data.length, "index out of range")
+	start, end = int64(a.offsets[i]), int64(a.offsets[i+1])
+	return
+}
+
+// LargeList represents an immutable sequence of array values.
+type LargeList struct {
+	array
+	values  arrow.Array
+	offsets []int64
+}
+
+// NewLargeListData returns a new LargeList array value, from data.
+func NewLargeListData(data arrow.ArrayData) *LargeList {
+	a := &LargeList{}
```

Comment:
Small consistency thing, use `new(LargeList)` like you do for `o := new(strings.Builder)`.